sdk/identity/Azure.Identity/src/UsernamePasswordCredential.cs

```diff
@@ -185,17 +194,19 @@ private async Task<AuthenticationRecord> AuthenticateImplAsync(bool async, Token
         private async Task<AccessToken> GetTokenImplAsync(bool async, TokenRequestContext requestContext, CancellationToken cancellationToken)
         {
             using CredentialDiagnosticScope scope = _pipeline.StartGetTokenScope("UsernamePasswordCredential.GetToken", requestContext);
-
             try
             {
-                var tenantId = TenantIdResolver.Resolve(_tenantId, requestContext, _allowMultiTenantAuthentication);
-
-                AuthenticationResult result = await Client
-                    .AcquireTokenByUsernamePasswordAsync(requestContext.Scopes, requestContext.Claims, _username, _password, tenantId, async, cancellationToken)
-                    .ConfigureAwait(false);
-
-                _record = new AuthenticationRecord(result, _clientId);
-
+                AuthenticationResult result;
+                if (_record != null)
+                {
+                    var tenantId = TenantIdResolver.Resolve(_tenantId, requestContext, _allowMultiTenantAuthentication);
+                    result = await Client.AcquireTokenSilentAsync(requestContext.Scopes, requestContext.Claims, _record, tenantId, async, cancellationToken)
```

Comment:
I think this should probably handle `MsalUiRequiredException` and call `AuthenticateImplAsync`, as I think this will be thrown when the refresh token has expired.

Reply:
handled, and also logged an event to indicate that silent auth failed.